rebar.config

```diff
@@ -1,20 +1,130 @@
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
 {minimum_otp_vsn, "R16"}.
 
-{erl_opts, [warnings_as_errors,
-            {parse_transform, lager_transform},
-            debug_info,
-            warn_untyped_record]}.
+{brt_protect, true}.
+
+{erl_opts, [
+    warn_export_all,
+    warn_export_vars,
+    warn_obsolete_guard,
+    warn_untyped_record,
+    warn_unused_import,
+    warnings_as_errors,
+    {parse_transform, lager_transform}
+]}.
 
-{deps, [getopt, lager]}.
+{escript_emu_args,
+    "%%! -escript main cuttlefish_escript +S 1 +A 0\n"}.
 
-{escript_emu_args, "%%! -escript main cuttlefish_escript +S 1 +A 0\n"}.
 {escript_incl_apps, [goldrush, getopt, lager, cuttlefish]}.
 {escript_main_app, cuttlefish}.
 
-{provider_hooks, [{post, [{compile, {default, escriptize}}]}]}.
+{provider_hooks, [
+    {pre,   [{compile,  neotoma}]}
+]}.
+
+{deps, [
+    {getopt,
+        {git, "https://github.com/basho/getopt.git",
+        {branch, "rebar3"} }},
+    {lager,
+        {git, "https://github.com/erlang-lager/lager.git",
+        {tag, "3.5.1"} }}
+]}.
+
+{profiles, [
+    {check, [
+        {dialyzer, [
+            {warnings, [
+                error_handling,
+                race_conditions,
+                unknown
+                % it might be good to clean these up at some point
+                % unmatched_returns
+            ]}
+        ]},
+        {erl_opts, [
+            debug_info,
+            {d, 'BASHO_CHECK'}
+        ]},
+        {overrides, [
+            {add, getopt,   [{erl_opts, [debug_info]}]},
+            {add, lager,    [{erl_opts, [debug_info]}]}
+        ]},
+        {xref_checks, [
+            deprecated_function_calls,
+            deprecated_functions,
+            locals_not_used,
+            undefined_function_calls,
+            undefined_functions
+        ]}
+    ]},
+    {docs, [
+        {edoc_opts, [
+            {app_default, "http://www.erlang.org/doc/man"},
+            {preprocess, true},
+            {macros, [
+                {'EDOC', true},
+                {'BASHO_DOCS', true}
+            ]}
+        ]}
+    ]},
+    {prod, [
+        {erl_opts, [
+            {d, 'BASHO_PROD'}
+        ]}
+    ]},
+    {test, [
+        {cover_enabled, true},
+        {erl_opts, [
+            debug_info,
+            nowarn_deprecated_function,
+            nowarn_export_all,
+            nowarn_unused_function,
+            nowarn_unused_import,
+            {d, 'BASHO_TEST'}
+        ]}
+    ]},
+    {validate, [
+        {erl_opts, [
+            strong_validation,
+            warn_missing_spec,
+            warn_untyped_record,
+            {d, 'BASHO_VALIDATE'}
+        ]}
+    ]}
+]}.
+
+{plugins, [
```

Comment:
what are the basho rebar tools? Where are they used that wasn't required before?